src/lib.udl

```diff
@@ -245,6 +245,90 @@ dictionary SignedAccountCredential {
     record<u8, string> signatures_hex;
 };
 
+[Custom]
+typedef bytes Bytes;
+
+/// An encrypted amount input to transfer from
+dictionary InputEncryptedAmount {
+    /// The aggregated encrypted amount.
+    Bytes agg_encrypted_amount;
+    /// The plaintext corresponding to the aggregated encrypted amount.
+    u64 agg_amount;
+    /// Index such that the `agg_amount` is the sum of all encrypted amounts on an account with indices strictly below `agg_index`.
+    u64 agg_index;
+};
+
+/// The payload data for a TransferToPublic transaction
+dictionary SecToPubTransferData {
+    /// The serialized remaining amount after deducting the amount to transfer
+    Bytes serialized_remaining_amount;
+    /// The amount to transfer
+    u64 transfer_amount;
+    /// The transfer index of the transfer
+    u64 index;
+    /// The serialized proof that the transfer is correct
+    Bytes serialized_proof;
+};
+
+dictionary SecToPubTransferDataDeserializeResult {
+    SecToPubTransferData value;
+    u64 bytes_read;
+};
+
+/// A credential with attributes, public keys, and proofs that it is
+/// well-formed.
+dictionary CredentialDeploymentInfo {
+    /// List of anonymity revokers which can revoke the identity.
+    /// The order is important since it is the same order as that signed by the identity provider,
+    /// and permuting the list will invalidate the signature from the identity provider.
+    record<u32, ChainArData> ar_data;
+    /// Credential registration ID of the credential serialized as hex.
+    Bytes cred_id;
+    /// Credential keys (i.e. account holder keys).
+    CredentialPublicKeys credential_public_keys;
+    /// Identity of the identity provider who signed the identity object
+    /// from which this credential is derived.
+    u32 ip_identity;
+    /// Policy of this credential object.
+    Policy policy;
+    /// The proofs of ownership serialized as hex
+    Bytes proofs;
```

Comment:
The value includes both fields serialized according to the `Serial` implementation of the corresponding proof type. 

The way I see it we'll never use these proofs for anything other that submitting them. If we were to ever verify the proofs in swift, this would be through FFI as well where we'd have to deserialize into the original construct from concordium-base.

Reply:
It just uses the `serde::Serialize` (derived as `SerdeBase16IgnoreLengthSerialize`) implementation of the proof struct from concordium-base to represent the proofs.

In general, the strategy is to define a struct which describes the JSON representation of the underlying struct from concordium-base.